### DISCUSSION TITLE
SIGSEGV while writing BMP image

### DISCUSSION THREAD

**Discussion by kstanikviacbs** [libvips/libvips#3071] [Q&A]
Hi again!

Creating an issue as I get SIGSEGV while unit testing outputting BMP file. Environment is quite specific - I utilise library based on GraphicsMagick for the source of truth while comparing images and on the other hand - have ImageMagick installed for vips as well. Mentioning that at the beginning since this might be related to root cause of the issue as trace below shows:

```
#0  0x00007f7cf011d49c in  () at /usr/lib/libc.so.6
#1  0x00007f7cf00cd958 in raise () at /usr/lib/libc.so.6
#2  0x00007f7cf00b75c7 in abort () at /usr/lib/libc.so.6
#3  0x00007f7cf0376e2d in  () at /usr/lib/libGraphicsMagick.so.3
#4  0x00007f7cf00cda00 in <signal handler called> () at /usr/lib/libc.so.6
#5  0x00007f7cf036ee7b in IsEventLogging () at /usr/lib/libGraphicsMagick.so.3
#6  0x00007f7cf0374e2f in LogMagickEventList () at /usr/lib/libGraphicsMagick.so.3
#7  0x00007f7cf0375aaa in LogMagickEvent () at /usr/lib/libGraphicsMagick.so.3
#8  0x00007f7cf03ba1f2 in SetClientName () at /usr/lib/libGraphicsMagick.so.3
#9  0x00007f7cef4dcb38 in  () at /usr/lib/libMagickCore-7.Q16HDRI.so.10
#10 0x00007f7cef4e30e1 in MagickCoreGenesis () at /usr/lib/libMagickCore-7.Q16HDRI.so.10
#11 0x00007f7cf075c1b0 in magick_genesis_cb () at /usr/lib/libvips.so.42
#12 0x00007f7cf0614277 in g_once_impl () at /usr/lib/libglib-2.0.so.0
#13 0x00007f7cf075ee14 in vips_foreign_save_magick_build () at /usr/lib/libvips.so.42
#14 0x00007f7cf075f8ab in vips_foreign_save_magick_buffer_build () at /usr/lib/libvips.so.42
#15 0x00007f7cf08e6767 in vips_object_build () at /usr/lib/libvips.so.42
#16 0x00007f7cf08f2a88 in vips_cache_operation_buildp () at /usr/lib/libvips.so.42
#17 0x00007f7cf08f9939 in vips_call_required_optional () at /usr/lib/libvips.so.42
#18 0x00007f7cf08fa558 in vips_call_split_option_string () at /usr/lib/libvips.so.42
#19 0x00007f7cf08ed79f in vips_image_write_to_buffer () at /usr/lib/libvips.so.42
```

As I understand it - runtime execution seems to be mixing ImageMagick and GraphicsMagick usage if they co-exist on the machine (switch between `#9` and `#8`), right? Is this something by design and vips assume they shouldn't be both installed or there's something to address here? Any help, clarification, guidance would be really appreciated. 

Thanks in advance!

**Environment**
 - OS: Arch Linux
 - Vips: 8.13.x (build from source, but with default magick-related options)

**Comment by jcupitt** [comment]
Hi @kstanikviacbs,

Ouch! Yes, mixing imagemagick and graphicsmagick sounds very bad, I'd be amazed if that didn't crash.

From your stacktrace it looks like libvips is calling into IM, but that is then calling into GM, which should not be possible. Perhaps your `LD_LIBRARY_PATH` needs checking? Does IM work at the command line?
